corehq/messaging/scheduling/scheduling_partitioned/migrations/0008_track_attempts.py

```diff
@@ -0,0 +1,53 @@
+# Generated by Django 2.2.24 on 2021-11-19 14:36
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('scheduling_partitioned', '0007_index_cleanup'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='alertscheduleinstance',
+            name='attempts',
+            field=models.IntegerField(default=0),
```

Comment:
Will this populate the existing entries with 0 or null?
if its 0, I am guessing that can be confusing and we would want them to be blank.

Reply:
It will and that was so that existing rows that have yet to be queued, will be treated properly. I can probably add that handling into the celery task, but I don't think there is a huge benefit. For existing records, those that have already been processed will look as though they were processed on the first try (since `attempts` is only incremented on failure), and those that have not been attempted will be in the correct state. So beyond being slightly misleading about how many tries a past record took, we can safely treat those numbers as correct in terms of handling the records.

Reply:
> For existing records, those that have already been processed will look as though they were processed on the first try

and would have attempts set to 0. May be then this should be named as failed_attempts?

>  I don't think there is a huge benefit

I was just thinking of someone looking at records, unaware that this column was added at a later stage, seeing attempts as 0 but more than one corresponding Logged Events and getting totally confused. If it was null it would still hint that this value is not expected to be present. I know this is unlikely but would be good to consider to keep data clean.

Reply:
(this is non blocking)